File: apps/web/types/market.ts
export type MarketSessionState = "open" | "closed" | "pre-open";

export interface MarketStatus {
  exchange: string;
  label: string;
  state: MarketSessionState;
  asOf: string;
  opensAt: string;
  closesAt: string;
}

export interface MarketIndex {
  symbol: string;
  name: string;
  value: number;
  change: number;
  percentChange: number;
}

File: apps/api/app/schemas/market.py
from typing import Literal

from pydantic import BaseModel


class MarketStatus(BaseModel):
    exchange: str
    label: str
    state: Literal["open", "closed", "pre-open"]
    as_of: str
    opens_at: str
    closes_at: str


class MarketIndex(BaseModel):
    symbol: str
    name: str
    value: float
    change: float
    percent_change: float

